sqlmesh/core/schema_diff.py

```diff
@@ -14,83 +17,366 @@ class SchemaDeltaOp(Enum):
     DROP = auto()
     ALTER_TYPE = auto()
 
+    @property
+    def is_add(self) -> bool:
+        return self == SchemaDeltaOp.ADD
+
+    @property
+    def is_drop(self) -> bool:
+        return self == SchemaDeltaOp.DROP
+
+    @property
+    def is_alter_type(self) -> bool:
+        return self == SchemaDeltaOp.ALTER_TYPE
+
+
+class ParentColumns(PydanticModel):
+    columns: t.List[t.Tuple[str, exp.DataType]]
+
+    @classmethod
+    def create(cls, *columns_and_types: t.Tuple[str, t.Union[str, exp.DataType]]) -> ParentColumns:
+        columns_and_data_types = [
+            (column_name, exp.DataType.build(column_type))
```

Comment:
is the str here dialect agnostic?

Reply:
Yeah so I was thinking if someone wanted to make sure their types were parsed with a specific dialect then they would just pass in a DataType object where they used that dialect.